daemon/emer-daemon.c

```diff
@@ -1665,3 +1866,74 @@ emer_daemon_get_permissions_provider (EmerDaemon *self)
   EmerDaemonPrivate *priv = emer_daemon_get_instance_private (self);
   return priv->permissions_provider;
 }
+
+gboolean
+emer_daemon_start_aggregate_timer (EmerDaemon          *self,
+                                   EmerAggregateTimer  *timer,
+                                   guint32              unix_user_id,
+                                   GVariant            *event_id,
+                                   GVariant            *aggregate_key,
+                                   gboolean             has_payload,
+                                   GVariant            *payload,
+                                   GError             **error)
+{
+  EmerDaemonPrivate *priv;
+  g_autoptr(EmerAggregateTimerImpl) timer_impl = NULL;
+  GVariant *nullable_payload;
+
+  g_return_val_if_fail (EMER_IS_DAEMON (self), FALSE);
+  g_return_val_if_fail (event_id != NULL, FALSE);
+  g_return_val_if_fail (aggregate_key != NULL, FALSE);
+
+  priv = emer_daemon_get_instance_private (self);
+
+  if (!priv->recording_enabled)
+    {
+      g_dbus_interface_skeleton_unexport (G_DBUS_INTERFACE_SKELETON (timer));
+      g_set_error (error, G_DBUS_ERROR, G_DBUS_ERROR_FAILED,
+                   "Metrics are disabled");
+      return FALSE;
+    }
+
+  if (!is_uuid (event_id))
+    {
+      g_dbus_interface_skeleton_unexport (G_DBUS_INTERFACE_SKELETON (timer));
+      g_warning ("Event ID must be a UUID represented as an array of %"
+                 G_GSIZE_FORMAT " bytes. Dropping event.", UUID_LENGTH);
+      g_set_error (error, G_DBUS_ERROR, G_DBUS_ERROR_FAILED,
+                   "Event ID must be a UUID represented as an array of %"
+                   G_GSIZE_FORMAT " bytes. Dropping event.",
+                   UUID_LENGTH);
+      return FALSE;
+    }
+
+  nullable_payload = get_nullable_payload (payload, has_payload);
+  timer_impl = emer_aggregate_timer_impl_new (priv->aggregate_tally,
+                                              timer,
+                                              unix_user_id,
+                                              event_id,
+                                              aggregate_key,
+                                              nullable_payload,
+                                              g_get_monotonic_time ());
+  g_object_set_data_full (G_OBJECT (timer_impl),
+                          "daemon",
+                          g_object_ref (self),
+                          g_object_unref);
+
+  if (g_hash_table_contains (priv->aggregate_timers, timer_impl))
+    {
+      g_warning ("There already is a timer for this event and user");
+      g_set_error (error, G_DBUS_ERROR, G_DBUS_ERROR_FAILED,
+                   "There already is a timer for this event and user");
```

Comment:
event, user, aggregate_key and payload.

I don't think we need the warning. we are returning an error to the caller.

```suggestion
      g_set_error (error, G_DBUS_ERROR, G_DBUS_ERROR_FAILED,
                   "There already is a running timer for this set of arguments");
```